File: src/MinStack.java
import java.util.Stack;

public class MinStack {
    private Stack<Integer> stack;
    private Stack<Integer> minStack;
    public MinStack() {
        stack = new Stack<>();
        minStack = new Stack<>();
    }

    public void push(int val) {
        stack.push(val);
        if (minStack.isEmpty()){
            minStack.push(val);
        }else if (minStack.peek()>val){
            minStack.push(val);
        }
    }

    public void pop() {
        if (stack.isEmpty())
            return;
        int val = stack.pop();
        if (minStack.peek()==val)
            minStack.pop();
    }

    public int top() {
        if (stack.isEmpty())
            return -1;
        return stack.peek();
    }

    public int getMin() {
        if (minStack.isEmpty())
            return -1;
        return minStack.peek();
    }
}

File: minStack.cpp
class MinStack {
    stack<int> s;
    int mn=INT_MAX;
public:
    MinStack() {
        
    }
    
    void push(int val) {
        if(mn>=val) {
            s.push(mn);
            mn=val;
        }
        s.push(val);
    }
    
    void pop() {
        if(s.top()==mn) {
            s.pop();
            mn=s.top();
        }
        s.pop();
    }
    
    int top() {
        return s.top();
    }
    
    int getMin() {
        return mn;
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(val);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */